app/models/placement.rb

```diff
@@ -47,4 +55,13 @@ def self.order_by_school_ids(school_ids)
     end
     order(condition)
   end
+
+  # TODO: Remove after data migrated
+  def assign_subject
+    return if subjects.blank? ||
+      subject.present? ||
+      subject == subjects.last
```

Comment:
`subjects.exists?` might be more performant here?

```suggestion
    return if subjects.exists? ||
      subject.present? ||
      subject == subjects.last
```

We could also predicate this `after_save` on a Proc instead of returning inside the method? Not a huge deal though :)

Reply:
I was thinking along the lines this: [9.2 Using :if and :unless with a Proc](https://guides.rubyonrails.org/active_record_callbacks.html#using-if-and-unless-with-a-proc)